### PR TITLE
DOCSP-13861 replace usage example

### PR DESCRIPTION
**Pull request by kyuan-mongodb**
## Pull Request Info

Added Usage Examples > Write Operations > Replace a Document page

### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-13861

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/golang/docsworker-xlarge/DOCSP-13861-ReplaceUsageExample/usage-examples/replaceOne/

### Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Does it render on staging correctly?
- [x] Are all the links working?
- [x] Are the staging links in the PR description updated?

### CODE REVIEW

File: source/includes/usage-examples/code-snippets/replace.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	if err != nil {
+		panic(err)
+	}
+
+	// When you run this file for the first time, it should print: "Matched and replaced an existing document."
+
+	if result.MatchedCount != 0 {
+		fmt.Println("Matched and replaced an existing document.")
+		return
+	}
```
Comment (terakilobyte) [inline source/includes/usage-examples/code-snippets/replace.go:48]: Is there a `ModifiedCount` field in the struct? If so, let's use that as it'd be more truthful that something was written.

File: source/usage-examples/replaceOne.txt [hunk 2]
```diff
@@ -3,3 +3,53 @@ Replace a Document
...
 .. default-domain:: mongodb
+
+You can replace a document in a collection by using the ``ReplaceOne()``
+method. 
+
+The following example specifies a query filter and replacement document
+to the ``ReplaceOne()`` method, which matches a document in the
+``movies`` collection where the ``title`` field is "Shrek" and replaces
+it with a document that contains a ``title`` and ``plot`` field about
+the movie:
```
Comment (terakilobyte) [inline source/usage-examples/replaceOne.txt:14]: I think we should stick with haikus now. There are a lot of other fields in a `movie` document that we don't want to show, nor overwrite.

**Review by terakilobyte — CHANGES_REQUESTED** [review]
left a suggestion that will require other changes, but it looks good!

**Review by terakilobyte — APPROVED** [review]
lgtm

**Review by kevinAlbs — APPROVED** [review]
Minor comment, but otherwise LGTM!

File: source/includes/usage-examples/code-snippets/replace.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+	if err != nil {
+		panic(err)
+	}
+
+	// When you run this file for the first time, it should print: "Number of documents replaced: 1"
+
+	if result.MatchedCount != 0 {
+		fmt.Println("Number of documents replaced: %d\n", result.ModifiedCount)
+		return
```
[the remark this answers is not in the document]
Comment (kyuan-mongodb) [reply source/includes/usage-examples/code-snippets/replace.go]: good catch, I'm not sure why I initially had it there

**Pull request merged**